pkg/apis/monitoring/v1/types.go

```diff
@@ -1605,6 +1605,16 @@ type AlertmanagerSpec struct {
 	// This is an alpha field and requires enabling StatefulSetMinReadySeconds feature gate.
 	// +optional
 	MinReadySeconds *uint32 `json:"minReadySeconds,omitempty"`
+	// GlobalAlertmanagerConfig used to set the global alertmanager config.
+	// The specified global alertmanager config will not force add a namespace label in routes and inhibitRules.
+	GlobalAlertmanagerConfig *GlobalAlertmanagerConfig `json:"globalAlertmanagerConfig,omitempty"`
+}
+
+// GlobalAlertmanagerConfig used to set the global alertmanager config.
+// +k8s:openapi-gen=true
+type GlobalAlertmanagerConfig struct {
+	// Name specifies the global AlertmanagerConfig name which must be in the same namespace as Alertmanager.
+	Name string `json:"name,omitempty"`
```

Comment:
name shouldn't be empty
```suggestion
	// +kubebuilder:validation:MinLength=1
	Name string `json:"name,omitempty"`
```